optaplanner-examples/src/main/java/org/optaplanner/examples/flightcrewscheduling/optional/score/FlightCrewSchedulingConstraintProvider.java

```diff
@@ -0,0 +1,63 @@
+package org.optaplanner.examples.flightcrewscheduling.optional.score;
+
+import java.util.Objects;
+
+import org.optaplanner.core.api.score.buildin.hardsoftlong.HardSoftLongScore;
+import org.optaplanner.core.api.score.stream.Constraint;
+import org.optaplanner.core.api.score.stream.ConstraintFactory;
+import org.optaplanner.core.api.score.stream.ConstraintProvider;
+import org.optaplanner.examples.flightcrewscheduling.domain.Employee;
+import org.optaplanner.examples.flightcrewscheduling.domain.FlightAssignment;
+
+public class FlightCrewSchedulingConstraintProvider implements ConstraintProvider {
+
+    @Override
+    public Constraint[] defineConstraints(ConstraintFactory constraintFactory) {
+        return new Constraint[]{
+                flightConflict(constraintFactory),
+                requiredSkill(constraintFactory),
+                employeeUnavailability(constraintFactory),
+                transferBetweenTwoFlights(constraintFactory),
+                firstAssignmentDepartingFromHome(constraintFactory),
+                lastAssignmentArrivingAtHome(constraintFactory)
+        };
+    }
+
+    private Constraint flightConflict(ConstraintFactory constraintFactory) {
+        return constraintFactory.fromUniquePair(FlightAssignment.class)
+                .filter((first, second) -> Objects.equals(first.getEmployee(), second.getEmployee()))
+                .filter((first, second) -> second.getFlight().overlaps(first.getFlight()))
+                .filter((first, second) -> second.getId() > first.getId())
```

Comment:
fromUniquePair automaticaly includes the "id > id" check, the "unique" signals that.
fromUniquePair is syntatic sugar for a join with an id comparison (see javadoc)